src/firefly/js/core/MasterSaga.js

```diff
@@ -74,7 +84,10 @@ export function* masterSaga() {
             case ADD_SAGA: {
                 const {getState, dispatch}= flux.getRedux();
                 const {saga,params}= action.payload;
-                if (typeof saga === 'function') yield fork( saga, params, dispatch, getState);
+                // with fork every exception will bubble up from the child to the parent:
+                // an unhandled exception in one saga will cancel all sibling sagas
+                // with spawn, only the saga with the unhandled error will be cancelled
+                if (typeof saga === 'function') yield spawn(saga, params, dispatch, getState);
```

Comment:
I know this was here before but I has actually bitten me.  We should `console.log` an error. Also why not us lodash `isFunction`?

Another question: I though  we had decided to use `fork` so we could catch it and log an error? Will spawn log and error?

Reply:
We decided that we need to use fork for action watchers, because the parent is catching errors. 

In this case, the individual sagas are responsible for catching and logging errors. The unhandled errors are caught by middleware. I have tested that with spawn the errors are still printed to console, but do not cancel the siblings. The only saga that is cancelled is the one that caused the unhandled error.